lib/omnibus/packagers/windows_base.rb

```diff
@@ -90,22 +90,84 @@ def thumbprint
       signing_identity[:thumbprint]
     end
 
-    def algorithm
-      signing_identity[:algorithm]
-    end
-
     def cert_store_name
       signing_identity[:store]
     end
 
-    def timestamp_servers
-      signing_identity[:timestamp_servers]
-    end
-
     def machine_store?
       signing_identity[:machine_store]
     end
 
+    def signing_identity_file(pfxfile = NULL, params = NULL)
+      unless null?(pfxfile)
+        @signing_identity_file = {}
+        unless pfxfile.is_a?(String)
+          raise InvalidValue.new(:pfxfile, "be a String")
+        end
+
+        @signing_identity_file[:pfxfile] = pfxfile
+
+        if !null?(params)
+          unless params.is_a?(Hash)
+            raise InvalidValue.new(:params, "be a Hash")
+          end
+
+          valid_keys = [:password, :timestamp_servers, :algorithm]
+          invalid_keys = params.keys - valid_keys
+          unless invalid_keys.empty?
+            raise InvalidValue.new(:params, "contain keys from [#{valid_keys.join(', ')}]. "\
+                                   "Found invalid keys [#{invalid_keys.join(', ')}]")
+          end
+
+          if params[:password].nil? 
+            raise InvalidValue.new(:params, "Must supply password for PFX file")
+          end
+        else
+          params = {}
+        end
+
+        @signing_identity_file[:algorithm] = params[:algorithm] || "SHA1"
+        servers = params[:timestamp_servers] || DEFAULT_TIMESTAMP_SERVERS
+        @signing_identity_file[:timestamp_servers] = [servers].flatten
+        @signing_identity_file[:password] = params[:password] || false
+        end
+
+        @signing_identity_file
+    end
+    expose :signing_identity_file
+
+    def pfx_algorithm
+      signing_identity_file[:algorithm]
+    end
+
+    def pfx_password
+      signing_identity_file[:password]
+    end
+
+    def pfx_file
+      signing_identity_file[:pfxfile]
+    end
+
+    def timestamp_servers
+      if signing_identity
```

Comment:
if `signing_identity` and `signing_identity_file` should be mutually exclusive, maybe it would just be safer to raise an exception if both are defined? I feel like there might be some room here for the wrong params to be used unexpectedly in `sign_package`... what do you think?

Reply:
They should be mutually exclusive, but the general pattern I've used when adding "features" is to allow the original to be the "default".  So, if someone supplies both, in this case, then `signing_identity` will take precedence.    That said, your suggestion makes tons of sense, and I'd be happy to make that change if you think it's worth it.

Reply:
I went ahead and made them mutually exclusive